### PR TITLE
feat: Integrate online events

### LINKED ISSUE #4611: Basic Event Details: Enable option Online Events or Mixed Event
With an increasing number of online events that don't have a location, we should also offer the possibility to run events without a location or event with a location and an online presence (mixed events).

Therefore please change the basic details wizard as follows:
* create 4 buttons
  * Venue
  * Online event
  * Mixed event
  * To be announced

Screenshot from 2020-08-09 06-06-43

When the user clicks the buttons:
- [x] Location - keep options as is
- [x] Online event, add following and show this on the public event page:
  - [x] Show a field in section online events for "Live streaming (Optional): https://________________________"
  - [x] Show a field in section online event for "Webinar (Optional): https://____________________________"
  - [ ] Show a tick box "Details will be shared as follows (e.g. you can share links in the session info: _Add__message___here." If the user chooses this message, display this message if the user on the public event page.
- [x] Mixed event: Show both the form for Location and Online event.
- [x] To be announced: Show the message "Details to be announced." on the public event page in the respective area.

Screenshot from 2020-07-22 18-35-27

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #4611

### CODE REVIEW

File: app/models/event.js [hunk 1]
```diff
@@ -154,6 +158,8 @@ export default class Event extends ModelBase.extend(CustomPrimaryKeyMixin, {
   endsAtTime   : computedDateTimeSplit.bind(this)('endsAt', 'time'),
 
   segmentedExternalEventUrl: computedSegmentedLink.bind(this)('externalEventUrl'),
+  segmentedLiveStreamUrl: computedSegmentedLink.bind(this)('liveStreamUrl'),
```
Comment (niranjan94) [inline app/models/event.js:161]: Codacy found an issue: [Missing space after key 'segmentedLiveStreamUrl'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6140897)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6140897&bid=20076068)

File: app/models/event.js [hunk 2]
```diff
@@ -154,6 +158,8 @@ export default class Event extends ModelBase.extend(CustomPrimaryKeyMixin, {
...
+  segmentedWebinarEventUrl: computedSegmentedLink.bind(this)('webinarUrl'),
```
Comment (niranjan94) [inline app/models/event.js:162]: Codacy found an issue: [Missing space after key 'segmentedWebinarEventUrl'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6140897)

**Comment by mariobehling** [comment]
New events seems to be working fine. Existing events show a line when choosing "Online".

E.g. compare https://open-event-frontend-git-fork-iamareebjamal-online-events.eventyay.vercel.app/events/16fa59c7/edit/basic-details

Screenshot from 2020-09-08 14-06-52

**Comment by iamareebjamal** [comment]
Thanks, will check.

**Comment by mariobehling** [comment]
It should be possible to publish an online event without adding URLs.
In the implementation of the PR it is not possible yet to publish an online event without adding URLs.

Screenshot from 2020-09-08 14-41-54

**Comment by iamareebjamal** [comment]
That might be a bug, but I have made it publishable without needing links. The work isn't complete since it was blocked by https://github.com/fossasia/open-event-server/pull/7251

**Comment by iamareebjamal** [comment]
@mariobehling Please check now. I have not yet completed `Show a tick box "Details will be shared as follows (e.g. you can share links in the session info: _Add__message___here." If the user chooses this message, display this message if the user on the public event page.`

I'll do it in another PR

**Review by mariobehling — APPROVED** [review]
Very good. I think remaining points can be done at a later time. Lets move forward.

**Pull request merged**